pkg/policy/rules.go

```diff
@@ -238,16 +238,14 @@ func (rules ruleSlice) updateEndpointsCaches(ep Endpoint, epSet *EndpointSet) (b
 	if securityIdentity == nil {
 		return false, fmt.Errorf("cannot update caches in rules for endpoint %d because it has a nil identity", id)
 	}
-
+	endpointSelected := false
 	for _, r := range rules {
+		// Update the matches cache of each rule, and note if
+		// the ep is selected by any of them.
 		if ruleMatches := r.matches(securityIdentity); ruleMatches {
-			epSet.Insert(ep)
-
-			// If epSet is updated, we can exit since updating it again if
-			// another rule selects the Endpoint is a no-op.
-			return true, nil
+			endpointSelected = true
```

Comment:
Can't we return earlier here?

Reply:
Functionally, yes, but the intent here is to update the caches of all the rules so that we can depend on the side-effects of the caches being up-to-date.

Reply:
oh... ok. This makes more sense. Thanks!